File: backend/src/models/comment.model.js
const mongoose = require("mongoose");

const ReplySchema = new mongoose.Schema({
  adminId: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
  content: { type: String, required: true, trim: true }
}, { timestamps: true });

const CommentSchema = new mongoose.Schema({
  productId: { type: mongoose.Schema.Types.ObjectId, ref: "Product", required: true },
  userId: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
  userName: { type: String, default: "" },
  content: { type: String, required: true, trim: true },
  rating: { type: Number, min: 1, max: 5 }, 
  
  // ✅ Mảng lưu trữ các đường dẫn ảnh (Ví dụ: ["/uploads/image1.jpg", "/uploads/image2.jpg"])
  images: { 
    type: [String], 
    default: [] 
  }, 
  
  isHidden: { type: Boolean, default: false },
  replies: { type: [ReplySchema], default: [] }
}, { timestamps: true });

// Sắp xếp index để truy vấn theo sản phẩm nhanh hơn (Tùy chọn)
CommentSchema.index({ productId: 1, createdAt: -1 });

module.exports = mongoose.model("Comment", CommentSchema);

File: Shop_giay/lib/features/comments/domain/comment_model.dart
class Reply {
  final String content;
  final DateTime createdAt;

  Reply({required this.content, required this.createdAt});

  factory Reply.fromJson(Map<String, dynamic> json) {
    return Reply(
      content: json['content'] ?? '',
      createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
    );
  }
}

class Comment {
  final String id;
  final String userName;
  final String content;
  final int? rating;
  final List<String> images; // ✅ THÊM BIẾN NÀY
  final DateTime createdAt;
  final List<Reply> replies;

  Comment({
    required this.id,
    required this.userName,
    required this.content,
    this.rating,
    required this.images,
    required this.createdAt,
    required this.replies,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['_id'] ?? '',
      userName: json['userName'] ?? 'Người dùng',
      content: json['content'] ?? '',
      rating: json['rating'],
      // ✅ PARSE MẢNG ẢNH TỪ JSON
      images: (json['images'] as List?)?.map((e) => e.toString()).toList() ?? [],
      createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
      replies: (json['replies'] as List?)?.map((e) => Reply.fromJson(e)).toList() ?? [],
    );
  }
}